common/crypto/platform/openssl/crypto.cpp

```diff
@@ -472,11 +571,17 @@ expected::ExpectedBool VerifySignData(
 	ret = EVP_PKEY_verify(
 		pkey_signer_ctx.get(), signature.data(), signature.size(), shasum.data(), shasum.size());
 	if (ret < 0) {
+		const string openssl_error_msg {GetOpenSSLErrorMessage()};
+		if (openssl_error_msg.find("asn1 encoding") != string::npos) {
```

Comment:
Oh man, is this the best way? They don't provide an error code for it?

I'm wondering if this might be unsafe. Imagine the scenario if openssl is upgraded to a version where this string has subtly changed. Then suddenly the client will not successfully validate any artifact anymore. And you don't know this until you try to do the *next* update. Maybe it's better if we always try?

Reply:
First thing I did was copy-paste the error code.

Guess what happened next :see_no_evil: :crying_cat_face: :sob: 

But I agree, maybe they have some error type or something

Reply:
In OpenSSL 3.x we get no error codes during verification, so I will fall back to ECDSA in case of all errors here.

Reply:
Looks good!